src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -949,42 +992,66 @@ namespace stan {
     boost::phoenix::function<validate_void_return_allowed>
     validate_void_return_allowed_f;
 
-    void validate_lhs_var_assgn::operator()(const std::string& name,
-                                       const scope& var_scope,
-                                       variable& v,  bool& pass,
-                                       const variable_map& vm,
-                                       std::ostream& error_msgs) const {
-      // validate existence
+    void validate_lhs_var_assgn::operator()(assgn& a,
+                                            const scope& var_scope,
+                                            bool& pass,
+                                            const variable_map& vm,
+                                            std::ostream& error_msgs) const {
+      std::string name = a.lhs_var_.name_;
+      if (!can_assign_to_lhs_var(name, var_scope, vm, error_msgs)) {
+        pass = false;
+        return;
+      }
+    }
+    boost::phoenix::function<validate_lhs_var_assgn>
+    validate_lhs_var_assgn_f;
+
+    void validate_lhs_var_assignment::operator()(variable_dims& v,
+                                                 const scope& var_scope,
+                                                 bool& pass,
+                                                 const variable_map& vm,
+                                                 std::ostream& error_msgs)
+      const {
+      std::string name = v.name_;
+      if (!can_assign_to_lhs_var(name, var_scope, vm, error_msgs)) {
+        pass = false;
+        return;
+      }
+    }
+    boost::phoenix::function<validate_lhs_var_assignment>
+    validate_lhs_var_assignment_f;
+
+    void validate_lhs_var_assgn_silent::operator()(const std::string& name,
```

Comment:
Is this silent in the sense that it does not produce any error messages? Then, why does it take `error_msgs` as an argument?

Reply:
Thanks for fixing!